src/forms/EditDocumentForm.js

```diff
@@ -17,98 +16,95 @@ const EditDocumentForm = ({
+  const [newFileName, setFileName] = useState();
+  const [newFileDescription, setFileDescription] = useState();
+  const [onUploading, setUploading] = useState(false);
+  const options = Object.keys(versionState).map(state => ({
+    key: state,
+    value: state,
+    text: versionState[state].title,
+    content: <Badge state={state} />,
+  }));
   return (
-    <form className="pt-16">
-      <GridContainer>
-        <label
-          className="Form--Label row-1 cell-3 text-right Form--Label-required"
-          htmlFor="file_name"
-        >
-          Document Title:
-        </label>
+    <Form
+      onSubmit={ev => {
+        if (handleSubmit) {
+          setUploading(true);
+          ev.preventDefault();
+          handleSubmit(newFileName, newFileType, newFileDescription);
+        }
+      }}
+    >
+      <Form.Field required>
+        <label>Document Title:</label>
         <input
           data-testid="name-input"
-          className="Form--Input border border-grey row-1 cell-9"
           type="text"
           name="file_name"
           placeholder="Phenotypic Data manifest for..."
           defaultValue={fileName}
-          onChange={e => onNameChange(e)}
+          onChange={e => {
+            onNameChange ? onNameChange(e) : setFileName(e.target.value);
+          }}
         />
-        <label
-          className="Form--Label row-2 cell-3 text-right"
-          htmlFor="file_name"
-        >
-          Approval Status:
-        </label>
-        <div className="row-2 cell-9">
-          <div className={dropdownClass}>
-            <button
-              type="button"
-              className="Dropdown--button justify-start pl-20"
-              tabIndex="0"
-              data-testid="status-dropdown"
-            >
-              <Badge state={versionStatus} />
-              <Icon
-                className="m-12"
-                width={20}
-                height={20}
-                kind="chevron-down"
-              />
-            </button>
-            <div className="Dropdown--container">
-              <ul className="Dropdown--list">
-                {versionStatusList.map(versionStatusValue => (
-                  <button
-                    onClick={() => onVersionStatusChange(versionStatusValue)}
-                    type="button"
-                    className="Dropdown--item"
-                    tabIndex="0"
-                    key={versionStatusValue}
-                  >
-                    <Badge state={versionStatusValue} />
-                  </button>
-                ))}
-              </ul>
-            </div>
-          </div>
-        </div>
-        <label
-          className="Form--Label row-3 cell-3 text-right Form--Label-required"
-          htmlFor="file_type"
-        >
-          Document Type:
-        </label>
-        <fieldset name="file_type" className="row-3 cell-9">
-          {Object.keys(fileTypeDetail).map(item => (
+      </Form.Field>
+      {versionStatus && (
```

Comment:
is this scoped to show only `Admin` role users?

Reply:
Will address in separate pr.